### PR TITLE
Improvement in HttpAuthorizationRequestHeader validation

### PR DESCRIPTION
**Pull request by eddynaka**
This PR fixes the following issues:
- example.com and example.com/ should return the same fingerprint
- adding a dynamic validator

### CODE REVIEW

File: Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs [hunk 1]
```diff
@@ -44,10 +55,63 @@ static HttpAuthorizationRequestHeaderValidator()
...
+            var fingerprint = new Fingerprint(fingerprintText);
+            string host = fingerprint.Host;
+
+            try
+            {
+                using HttpClient client = CreateHttpClient();
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Guid.NewGuid().ToString());
+
+                using HttpResponseMessage responseDummy = client
+                    .GetAsync(host + (fingerprint.Resource ?? string.Empty), HttpCompletionOption.ResponseHeadersRead)
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs:78]: >fingerprint.Resource [](start = 38, length = 20)

please lift this into its own variable. generally, it's helpful to show early in the validator what data from the fingerprint is used in the method. this approach also helps lead to more concise lines of code, as a variable named 'resource' is fewer columns than 'fingerprint.Resource'. Note that using a variable will allow you to handle the conditional assignment to string.empty as well, also shortening code. :)

**Comment by michaelcfanning** [comment]
>         public static string ReturnUnauthorizedAccess(ref string message,

Can we also include a regex fix in this change? Or are you doing that next?

---
Refers to: Src/Plugins/ValidatorBase.cs:166 in a521770. [](commit_id = a521770bf29bc2ca5b1032d5bb374141813eb6ee, deletion_comment = False)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Comment by eddynaka** [comment]
>         public static string ReturnUnauthorizedAccess(ref string message,

i was trying. but, my changes was always making catastrophic issues...will probably need some help to understand.

---
In reply to: [780101549](https://github.com/microsoft/sarif-pattern-matcher/pull/220#issuecomment-780101549) [](ancestors = 780101549)

---
Refers to: Src/Plugins/ValidatorBase.cs:166 in a521770. [](commit_id = a521770bf29bc2ca5b1032d5bb374141813eb6ee, deletion_comment = False)

File: Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs (hunk 1 above)
Comment (eddynaka) [reply Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs]: changed the code. Let me know what do you think :)

---
In reply to: [577118840](https://github.com/microsoft/sarif-pattern-matcher/pull/220#discussion_r577118840) [](ancestors = 577118840)

**Comment by eddynaka** [comment]
>         public static string ReturnUnauthorizedAccess(ref string message,

replaced \s for \n. that fixed the catastrophic error.

---
In reply to: [780102582](https://github.com/microsoft/sarif-pattern-matcher/pull/220#issuecomment-780102582) [](ancestors = 780102582,780101549)

---
Refers to: Src/Plugins/ValidatorBase.cs:166 in a521770. [](commit_id = a521770bf29bc2ca5b1032d5bb374141813eb6ee, deletion_comment = False)

**Pull request merged**